packages/xsnap/src/xsnap.js

```diff
@@ -132,14 +136,14 @@ export function xsnap(options) {
 
   /**
    * @param {string} code
-   * @returns {Promise<void>}
+   * @returns {Promise<Uint8Array | void>}
    */
   async function evaluate(code) {
     const result = baton.then(async () => {
       await messagesToXsnap.next(encoder.encode(`e${code}`));
-      await runToIdle();
+      return runToIdle();
```

Comment:
You mentioned in chat that something hurt your head. I bet this was it. 😕 Sorry. Using `await` case-by-case to avoid connecting the return type was convenient.